src/main/java/dev/mlml/command/Context.java

```diff
@@ -67,21 +68,46 @@ public static String[] split(Message message) {
     public boolean parse(Command command) {
         List<? extends ArgumentBase<?>> commandarguments = command.getArguments();
 
+        if (args == null || args.length == 0) {
+            // Handle the case where no arguments are given
+            for (ArgumentBase<?> arg : commandarguments) {
+                if (arg.isRequired()) {
+                    logger.debug("Required argument not found: {}", arg.getName());
+                    parsedArguments.invalidate();
+                    return true;
+                }
+                parsedArguments.add(arg);
+            }
+            return false; // No required arguments, parsing successful with no arguments
+        }
+
         int offset = 0;
         for (int i = 0; i < commandarguments.size(); i++) {
             ArgumentBase<?> arg = commandarguments.get(i);
             logger.debug("Parsing argument: {}, offset: {}", arg.getName(), offset);
             if (arg.getClass() == StringArgument.class && ((StringArgument) arg).isVArgs()) {
-                logger.debug("Variable argument found, skipping to end");
-                parsedArguments.add(arg, String.join(" ", List.of(args).subList(i + offset, args.length)));
+                logger.debug("Variable string argument found, skipping to end");
+                if (!List.of(args).subList(i + offset, args.length).isEmpty()) {
+                    parsedArguments.add(arg, String.join(" ", List.of(args).subList(i + offset, args.length)));
+                } else {
+                    parsedArguments.add(arg);
+                }
                 break;
             }
-            if (arg.isRequired() && i > args.length + i + offset) {
-                logger.debug("Required argument not found: {}, args.length: {}, offset: {}",
-                             arg.getName(),
-                             args.length,
-                             offset
-                );
+            if (i >= args.length + offset) {
+                if (arg.isRequired()) {
+                    logger.debug("Required argument not found: {}, args.length: {}, offset: {}",
+                                 arg.getName(),
+                                 args.length,
+                                 offset
+                    );
+                    parsedArguments.invalidate();
+                } else {
+                    for (int j = i + offset; j < commandarguments.size(); j++) {
+                        parsedArguments.add(arg);
+                    }
+                    logger.debug("Filled all empty arguments with default");
+                }
```

Comment:
`return true` is the fail condition
If you're backfilling all of the arguments, shouldn't `false` be returned?